### PR TITLE
docs: Rework syntax by example

### PR DESCRIPTION
**Pull request by juliaogris**
Rework syntax by example to match the current state of evy. Remove
references to builtin functions which will be treated in a separate
document.

---

There will be a bunch of follow up commits, next probably builtins,
then spec and README.md. I'm trying to keep things digestible here and just do one file at a time. Diff is probably a bit useless, 💰 

Maybe easiest reviewed on branch:
https://github.com/foxygoat/evy/blob/docs/docs/syntax_by_example.md

### CODE REVIEW

File: docs/syntax_by_example.md [hunk 1]
```diff
@@ -121,64 +117,69 @@ aspects of `evy`'s syntax. For a formal language specification see the
...
+            printf "%v, " arg
         end
+        printf "%v" args[-1]
     end
 
+### Function calls
+
+    n := add 1 2        // 3
+    foxprint "🐾"       // 🦊 🐾  
+    list 2 true "blue" // 2, true, blue
```
Comment (camh-) [inline docs/syntax_by_example.md:131]: nit: this comment does not line up with the ones above it

File: docs/syntax_by_example.md [hunk 2]
```diff
@@ -188,201 +189,47 @@ aspects of `evy`'s syntax. For a formal language specification see the
...
+## Event handling 
 
-    on mouse_down
-        print mouse_x mouse_y
+    on key
+        print "key pressed"
     end
 
-## Builtin
+Evy can only handle a limited set of events, such as keyboard presses,
```
Comment (camh-) [inline docs/syntax_by_example.md:227]: s/keyboard presses/key presses/

**Review by camh- — APPROVED** [review]
:green_salad:

File: docs/syntax_by_example.md (hunk 1 above)
Comment (juliaogris) [reply docs/syntax_by_example.md:131]: it lines up with on the output (I actually changed it).... I will change it back.

Comment (juliaogris) [reply docs/syntax_by_example.md:131]: done.

File: docs/syntax_by_example.md (hunk 2 above)
Comment (juliaogris) [reply docs/syntax_by_example.md:227]: done.

**Pull request merged**